reporter/reporter.go

```diff
@@ -213,6 +213,14 @@ func (t *reporter) reportInternal(span tracer.RawSpan) {
 	}
 }
 
+func (t *reporter) copyTag(oriTags map[string]string) map[string]string {
+	newTags := make(map[string]string)
```

Comment:
More memory efficient to allocate the map for the desired size: `make(map[string]string, len(oriTags) + 1)`. +1 is for the additional error tag you will be adding to the returned map.